rerun_py/rerun_sdk/rerun/components/quaternion.py

```diff
@@ -1,15 +1,25 @@
 from __future__ import annotations
 
+from dataclasses import dataclass
+
 import numpy as np
 import numpy.typing as npt
 import pyarrow as pa
 
 from rerun.components import REGISTERED_COMPONENT_NAMES, ComponentTypeFactory
 
-__all__ = [
-    "QuaternionArray",
-    "QuaternionType",
-]
+__all__ = ["QuaternionArray", "QuaternionType", "Quaternion"]
+
+
+@dataclass
+class Quaternion:
+    """3D rotation expressed via a Quaternion."""
+
```

Comment:
I think we want to make this a require argument by overriding the dataclass constructor
```suggestion
    def __init__(self, *, xyzw):            
        self.xyzw = xyzw
```